src/script/builder/tir/block_frame.h

```diff
@@ -62,15 +64,56 @@ class BlockFrameNode : public TIRFrameNode {
 };
 
 class BlockFrame : public TIRFrame {
+ public:
+  BlockFrame() = default;
+
  public:
   TVM_DEFINE_MUTABLE_NOTNULLABLE_OBJECT_REF_METHODS(BlockFrame, TIRFrame, BlockFrameNode);
 };
 
-BlockFrame Block_(String name);
+BlockFrame Block_(String name, bool no_realize = false);
+
+class BlockInitFrameNode : public TIRFrameNode {
+ public:
+  Optional<tvm::tir::Stmt> init;
```

Comment:
I don't think we need this field. Instead, when `ExitWithScope` is called, what we do is:

```python
# some error checking
if (!this->stmt.empty()) {
  # some error checking
  block = Find BlockFrame;
  block->init = AsStmt(this->stmts);
  # some error checking
}
```